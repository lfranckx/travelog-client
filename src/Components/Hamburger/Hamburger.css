/* HAMBURGER MENU STYLES */
:root {
    --primary-color: #287FB8;
    --overlay-color: rgba(2, 16, 28, 0.85);
    --menu-speed: 0.75s;
}

.menu-wrap {
    z-index: 1;
}

.menu-wrap .toggler {
    position: absolute;
    top: 10px;
    right: 5px;
    z-index: 2;
    cursor: pointer;
    width: 50px;
    height: 50px;
    opacity: 0;
}

.menu-wrap .hamburger {
    z-index: 1;
    width: 40px;
    height: 60px;
    display: flex;
    align-items: center;
    justify-content: center;
}

/* Hamburger Line */
.menu-wrap .hamburger > div {
    position: relative;
    flex: none;
    width: 100%;
    height: 3px;
    background:  #27374c;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all 0.4s ease;
}

/* Hamburger Lines - Top & Bottom */
.menu-wrap .hamburger > div::before,
.menu-wrap .hamburger > div::after {
    content: '';
    position: absolute;
    z-index: 1;
    top: -10px;
    width: 100%;
    height: 3px;
    background: inherit;
}

/* Moves Line Down */
.menu-wrap .hamburger > div::after {
    top: 10px;
}

/* Toggler Animation */
.menu-wrap .active + .hamburger > div {
    transform: rotate(135deg);
}

/* Turns Lines Into X */
.menu-wrap .active + .hamburger > div:before,
.menu-wrap .active + .hamburger > div:after {
    top: 0;
    transform: rotate(90deg);
}

/* Rotate On Hover When Checked */
.menu-wrap .active:hover + .hamburger > div {
    transform: rotate(225deg);
}

/* Show Menu */
.menu-wrap .active ~ .menu {
    visibility: visible;
}

.menu-wrap .active ~ .menu > div {
    transform: scale(1);
    transition-duration: var(--menu-speed);
}

.menu-wrap .active ~ .menu > div > div {
    opacity: 1;
    transition:  opacity 0.4s ease 0.4s;
}

.menu-wrap .menu {
    position: fixed;
    top: 0;
    right: 0;
    width: 100%;
    height: 100%;
    visibility: hidden;
    overflow: hidden;
    display: flex;
    align-items: center;
    justify-content: center;
}

.menu-wrap .menu > div {
    background: var(--overlay-color);
    border-radius: 50%;
    width: 200vw;
    height: 200vw;
    display: flex;
    flex: none;
    align-items: center;
    justify-content: center;
    transform: scale(0);
    transition: all 0.4s ease;
}

.menu-wrap .menu > div > div {
    text-align: center;
    max-width: 90vw;
    max-height: 100vh;
    opacity: 0;
    transition: opacity 0.4s ease;
}

.menu-wrap .menu > div > div > ul > li {
    list-style: none;
    font-size: 1.5rem;
    padding: 1rem;
}

.menu-wrap .menu > div > div > ul > li > a {
    color: #6ce4c4;
    
    transition: color 0.4s ease;
}

.close {
    display: none;
}

@media (min-width: 1001px) {
    .menu-wrap {
        display: none;
    }
}

@media (max-width: 500px) {
    .menu-wrap .hamburger {
        width: 30px;
        height: 45px;
    }
}