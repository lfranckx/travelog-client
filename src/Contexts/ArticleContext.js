/*eslint semi: ["error", "always"]*/
import React, { Component } from 'react';

export const nullArticle = {
    author: {},
    tags: [],
};
export const nullAuthor = {};
export const nullUser = {};
export const nullUsersArticles = [];

const ArticleContext = React.createContext({
    error: null,
    article: nullArticle,
    author: nullAuthor,
    user: nullUser,
    articlesList: [],
    authorsList: [],
    usersArticles: nullUsersArticles,
    setError: () => {},
    clearError: () => {},
    setArticle: () => {},
    clearArticle: () => {},
    setAuthor: () => {},
    clearAuthor: () => {},
    setUser: () => {},
    clearUser: () => {},
    setArticlesList: () => {},
    setAuthorsList: () => {},
    setUsersArticles: () => {},
    clearUsersArticles: () => {}
});
export default ArticleContext;

export class ArticleProvider extends Component {
    state = {
        error: null,
        article: nullArticle,
        author: nullAuthor,
        user: nullUser,
        articlesList: [],
        authorsList: [],
        usersArticles: []
    };

    setError = error => {
        console.error(error);
        this.setState({ error });
    };

    clearError = () => {
        this.setState({ error: null });
    };

    setArticle = article => {
        this.setState({ article });
    };

    clearArticle = () => {
        this.setArticle(nullArticle);
    };

    setAuthor = author => {
        this.setState({ author });
    };

    clearAuthor = () => {
        this.setAuthor(nullAuthor);
    };

    setUser = user => {
        console.log('setting user:', user);
        this.setState({ user });
    };

    clearUser = () => {
        this.setState(nullUser);
    };

    setArticlesList = articlesList => {
        this.setState({ articlesList });
    };

    setAuthorsList = authorsList => {
        this.setState({ authorsList });
    };

    setUsersArticles = usersArticles => {
        this.setState({ usersArticles });
    };

    clearUsersArticles = () => {
        this.setUsersArticles(nullUsersArticles);
    };

    render() {
        const value = {
            error: this.state.error,
            article: this.state.article,
            author: this.state.author,
            user: this.state.user,
            articlesList: this.state.articlesList,
            authorsList: this.state.authorsList,
            usersArticles: this.state.usersArticles,
            setError: this.setError,
            clearError: this.clearError,
            setArticle: this.setArticle,
            clearArticle: this.clearArticle,
            setAuthor: this.setAuthor,
            clearAuthor: this.clearAuthor,
            setUser: this.setUser,
            clearUser: this.clearUser,
            setArticlesList: this.setArticlesList,
            setAuthorsList: this.setAuthorsList,
            setUsersArticles: this.setUsersArticles,
            clearUsersArticles: this.clearUsersArticles
        };
        
        return (
            <ArticleContext.Provider value={value}>
                {this.props.children}
            </ArticleContext.Provider>
        );
    }   
}