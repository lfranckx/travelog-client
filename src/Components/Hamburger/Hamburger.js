import React, { Component } from 'react';
import './Hamburger.css';
import { Link } from 'react-router-dom';
import ArticleContext from '../../Contexts/ArticleContext';
import TokenService from '../../Services/token-service';
import IdleService from '../../Services/idle-service';


export default class Hamburger extends Component {
    static contextType = ArticleContext;

    handleLogOut = () => {        
        this.context.toggleActive();
        TokenService.clearAuthToken();
        TokenService.clearCallbackBeforeExpiry();
        IdleService.unRegisterIdleResets();
    }

    render() {
        const active = this.context.active;
        const { user } = this.props;
        const hamburgerMenu = (active) => (
            <div className={`menu-wrap`}>
                <div className={`toggler ${active ? "active" : ""}`}
                    onClick={() => {this.context.toggleActive()}}/>
                <div className="hamburger"><div></div></div>
                <div className={`menu`}>
                    <div>
                        <div>
                            <ul>
                                <li>
                                    <Link to='/' 
                                        onClick={() => {this.context.toggleActive()}}>
                                        Home
                                    </Link>
                                </li>
                                <li>
                                    <Link to="/post"
                                        onClick={() => {this.context.toggleActive()}}>
                                        Post Article
                                    </Link>
                                </li>
                                <li>
                                    <Link to={`/profile/${user.username}`}
                                        onClick={() => {this.context.toggleActive()}}>
                                        Profile
                                    </Link>
                                </li>
                                <li>
                                    <Link to="/"
                                        onClick={this.handleLogOut}>
                                        Logout
                                    </Link>
                                </li>
                            </ul>
                        </div>
                    </div>
                </div>
            </div>
        )
        return hamburgerMenu(active)
    }
}