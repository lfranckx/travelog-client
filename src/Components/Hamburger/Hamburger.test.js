import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Hamburger from './Hamburger';
import { ArticleProvider } from '../../Contexts/ArticleContext';

describe(`Hamburger component`, () => {
    const props = {
            username: 'user1',
            about: 'This is my about me.',
            name: 'Sam Smith',
            profile_image: 'http://placehold.it/500x500',
        };

    it(`renders page without crashing`, () => {
        const wrapper = shallow(
            <ArticleProvider >
                <Hamburger {...props}/>
            </ArticleProvider>
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    });
});