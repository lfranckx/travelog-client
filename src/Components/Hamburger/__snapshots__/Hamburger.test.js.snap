// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger component renders page without crashing 1`] = `
<ContextProvider
  value={
    Object {
      "active": false,
      "addComment": [Function],
      "article": Object {},
      "articlesList": Array [],
      "author": Object {},
      "authorsList": Array [],
      "clearArticle": [Function],
      "clearAuthor": [Function],
      "clearError": [Function],
      "clearUser": [Function],
      "clearUsersArticles": [Function],
      "comments": Array [],
      "error": null,
      "search": "",
      "setArticle": [Function],
      "setArticlesList": [Function],
      "setAuthor": [Function],
      "setAuthorsList": [Function],
      "setComments": [Function],
      "setError": [Function],
      "setUser": [Function],
      "setUsersArticles": [Function],
      "toggleActive": [Function],
      "updateSearch": [Function],
      "user": Object {},
      "usersArticles": Array [],
    }
  }
>
  <Hamburger
    about="This is my about me."
    name="Sam Smith"
    profile_image="http://placehold.it/500x500"
    username="user1"
  />
</ContextProvider>
`;
